src/components/Contact.js

```diff
@@ -39,19 +48,37 @@ class Contact extends React.Component {
   }
   
   get getAppropriateJSX() {
-    const { isFavourite } = this.props;
-    const { contact } = this.state;
+    const { isFavourite, contact } = this.props;
     if (!contact) return <InLineSpinner />;
```

Comment:
Instead of having all those 3 conditions I would do something else. Get rid of this function and have only the this.createNormalContactJSX function. Rename that to something like renderContact and have only one return with conditions in JSX. Here is an example:

render() {
  return (
      !contact ? <InLineSpinner /> : <div...................................... > 
    )
}

You can do the same for things that dont need to be displayed for the favourite contact. Here is an example:
<div>
  isFavourite ? 'write here code for favourite contact' : 'write here code if normal contact'
</div>

Here is some documentation of the conditional operator: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator